app/models/user.rb

```diff
@@ -5,4 +5,10 @@ class User < ActiveRecord::Base
          :recoverable, :rememberable, :trackable, :validatable, :confirmable
 
   has_many :topics
+  has_many :bookmarks
+  has_many :likes, dependent: :destroy
+
+  def liked(post)
+    likes.where(bookmark_id: bookmark.id).first
```

Comment:
This line of code is trying to reference the variable `bookmark` but it doesn't exist for `User`. If you notice the function parameter, that might give you the clue of what needs to change. :smiley: